tensorflow/compiler/xla/service/gpu/launch_dimensions.cc

```diff
@@ -81,12 +82,28 @@ StatusOr<LaunchDimensions> CalculateLaunchDimensions(
   // need more registers to hold intermediate values. Reduce the number of
   // blocks per thread to increase the number of registers available to ptxas.
   // Make sure we still have a multiple of 32.
-  threads_per_block =
+  int64 threads_per_block_row_optimized = shape.dimensions().back() / unroll_factor;
```

Comment:
I do not think it would make it more readable.
Here is the high level of the algo.
```
int64 threads_per_block = max_sizes;
if (row_vectorized && ...) {
threads_per_block = row_vectorized_threads_per_block;
} else {
threads_per_block = RoundUpToNearest32();
}
int64 block_count = CeilOfRatio(num_elements, threads_per_block);
if (few_waves) {
Update threads_per_block  and block_count.
} 
```

Currently, we can't really use few_waves and row_vectorized together.
I have a follow up task that is to combine both to get even higher speed up.
To keep this PR small enough, are you fine that I look possibly moving that to a function at that time?